Lib/asyncio/proactor_events.py

```diff
@@ -684,7 +690,13 @@ def close(self):
             return
 
         if threading.current_thread() is threading.main_thread():
-            signal.set_wakeup_fd(-1)
+            oldfd = signal.set_wakeup_fd(-1)
+            if oldfd == self._csock.fileno():
```

Comment:
Is this maybe backwards? In the unix code you use `!=`.

Separately, the same concern about warnings during closing as for the unix version.

And maybe the error message doesn't have to be split (it's not the end of the world if it's a little too long :-).

Reply:
> Is this maybe backwards? In the unix code you use !=.

Indeed :man_facepalming: Fixed in https://github.com/python/cpython/pull/96807/commits/bc7b849d3b29b643f0352441142e95fb0a72bcbe.